synfig-studio/src/gui/widgets/widget_waypoint.cpp

```diff
@@ -294,5 +289,62 @@ Widget_Waypoint::get_waypoint()const
 	return waypoint;
 }
 
+void
+Widget_Waypoint::config_tcb_params(bool show_params)
+{
+	if (show_params) {
+		// show labels
+		tcbFrame->get_label_widget()->show();
+		tensionLabel->show();
+		continuityLabel->show();
+		biasLabel->show();
+		temporalTensionLabel->show();
+		
+		// set the adjustment value
+		adj_tension->set_value(waypoint.get_tension());
+		adj_continuity->set_value(waypoint.get_continuity());
+		adj_bias->set_value(waypoint.get_bias());
+		adj_temporal_tension->set_value(waypoint.get_temporal_tension());
+		
+		// show spin buttons
+		spin_tension->show();
+		spin_continuity->show();
+		spin_bias->show();
+		spin_temporal_tension->show();
+	} else {
+		// hide labels
+		tcbFrame->get_label_widget()->hide();
+		tensionLabel->hide();
+		continuityLabel->hide();
+		biasLabel->hide();
+		temporalTensionLabel->hide();
+		
+		// hide spin buttons
+		spin_tension->hide();
+		spin_continuity->hide();
+		spin_bias->hide();
+		spin_temporal_tension->hide();
+	}
+}
+void
+Widget_Waypoint::on_before_options_changed()
+{
+	if (
+		(Waypoint::Interpolation)before_options->get_value() == Interpolation::INTERPOLATION_TCB ||
+		(Waypoint::Interpolation)after_options->get_value() == Interpolation::INTERPOLATION_TCB)
+		config_tcb_params(true);
+	else
+		config_tcb_params(false);
+}
 
+void
+Widget_Waypoint::on_after_options_changed()
+{
+	if (
+		(Waypoint::Interpolation)before_options->get_value() == Interpolation::INTERPOLATION_TCB ||
+		(Waypoint::Interpolation)after_options->get_value() == Interpolation::INTERPOLATION_TCB)
+		config_tcb_params(true);
+	else
+		config_tcb_params(false);
+}
```

Comment:
Avoid code repetition: maybe later someone touches the code but don't do it in both places…
So, I suggest to create a method named `update_tcb_params_visibility()` or something like that.

These callbacks can be or not be removed.

Reply:
I didn't notice I was implementing the same logic on those two signal functions.
Thank you!